File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController', function ($scope,brandService,$controller) {

    $controller('baseController',{$scope:$scope});//继承

    //查找全部
    $scope.findAll = function () {

        brandService.findAll().success(
            function (response) {
                $scope.list = response;

            }
        )
    }

    //分页查询
    $scope.findPage = function (pageNum, pageSize) {

        brandService.findPage(pageNum, pageSize).success(
            function (response) {
                $scope.rowList = response.rows;
                $scope.paginationConf.totalItems = response.total;
            }
        )
    }




    //新增,修改
    $scope.save = function () {
        //修改和新增公用一个编辑框,点击保存的时候,需要进行判断,是修改操作还是新增操作
        var object;
        if ($scope.entity.id != null) {
            object = brandService.update();

        }else{
            object = brandService.insert();

        }

        object.success(
            function (response) {
                if (response.success) {
                    $scope.reloadList();
                }
                else {
                    alert(response.message);
                }
            }
        )

    }
    //修改操作一.查询当前品牌信息

    $scope.findById = function (id) {

        $http.get("http://localhost:9101/brand/findById.do?id=" + id).success(
            function (response) {
                $scope.entity = response;
            }
        )
    }


    //删除操作步骤2:把id数组传给后台进行删除
    $scope.delete = function () {
        if(confirm("是否要删除")){
            brandService.delete($scope.ids).success(
                function (response) {
                    if (response.success) {
                        $scope.reloadList();
                        $scope.ids=[];//重置数组
                    }
                    else {
                        alert(response.message);
                    }
                }
            )
        }

    }

    //条件查询,取代原始分页查询
    $scope.conditions = {};
    $scope.search = function (pageNum, pageSize) {

        brandService.search(pageNum,pageSize,$scope.conditions).success(
            function (response) {


                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;


            }
        )
    }



})

File: pinyougou-manager-web/src/main/java/com/pinyougou/sellergoods/controller/BrandController.java
package com.pinyougou.sellergoods.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    //查询所有品牌
    @RequestMapping("findall")
    public List<TbBrand> findAll() {

        List<TbBrand> all = brandService.findAll();
        System.out.println("sucd");
        return all;


    }

    //分页查询
    @RequestMapping("page")
    public PageResult findPage(int pageNum, int pageSize) {
        PageResult pageResult = brandService.findPage(pageNum, pageSize);

        return pageResult;
    }

    //品牌新增
    @RequestMapping("insert")
    public Result insertBrand(@RequestBody TbBrand tbBrand) {


        try {
            brandService.insert(tbBrand);
            return new Result(true, "增加成功");
        } catch (Exception e) {

            e.printStackTrace();
            return new Result(false, "增加失败");
        }


    }

    //修改操作:1.查询
    @RequestMapping("findById")
    public TbBrand findById(Long id){
        TbBrand byId = brandService.findById(id);
            return byId;
    }

    //修改操作:2:接受修改后的数据,进行更新操作
    @RequestMapping("update")
    public Result update(@RequestBody TbBrand tbBrand){
        try {
            brandService.update(tbBrand);
            return new Result(true,"修改成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"修改失败");
        }

    }

    //删除操作
    @RequestMapping("delete")
    public Result delete(Long[] ids){

        try {
            brandService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    //条件查询
    @RequestMapping("search")
    public PageResult selectByCondition(@RequestBody TbBrand tbBrand,int pageNum,int pageSize){
        PageResult page = brandService.findPage(tbBrand, pageNum, pageSize);

        return page;

    }
    @RequestMapping("selectBrandList")
    public List<Map> selectBrandList(){
        return brandService.selectBrandList();
    }
}
